app/service/RedirectService.scala

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package service
+
+import models._
+import models.request.AuthenticatedCaseRequest
+import play.api.mvc.ControllerHelpers._
+import play.api.mvc.Results.Redirect
+import play.api.mvc._
+
+import javax.inject.Inject
+
+class RedirectService @Inject() () {
+
+  def redirectApplication(reference: String, fileId: Option[String] = None)(
```

Comment:
I know this is somewhat tested in the other controller specs, but could we get a test spec for this covering each case?